src/main/scala/chiseltest/internal/GenericBackend.scala

```diff
@@ -219,8 +221,11 @@ class GenericBackend[T <: Module](
           thread.thread.interrupt()
         }
       }
-
-      tester.finish() // needed to dump VCDs + terminate any external process
+      try {
+        tester.finish() // needed to dump VCDs + terminate any external process
+      } catch {
+        case e: TestApplicationException =>  throw new ChiselAssertionError(s"Simulator exited sooner than expected. See logs for more information about what is assumed to be a Chisel Assertion which failed.")
```

Comment:
i wasn't sure how to determine. It would appear that VCS returns 0 even if `assert` fires. Talking to some folks here i was told its not common to look at VCS error code to determine if an assertion failed. SO this would involve parsing log messages??

Reply:
I am not sure what the best approach would be. Verilator allows you to hook into the assertion mechanism. Not sure if there is a similar API for VPI. This would be cool future work, but of course nothing that will hold up this PR.